### PR TITLE
Closes #15

### LINKED ISSUE #15: Implement Shape for Cube
Implement the Shape trait for the Cube structure. The Cube structure has to be created aswell.

### PR DESCRIPTION
**Pull request by coffecup25**
Implement Shape and BoundingVolume for Cube. Tests are included.

### CODE REVIEW

**Review by isakvlarsson — APPROVED** [review]

File: pphysx3d/src/shapes/bounding_volume.rs [hunk 1]
```diff
@@ -1,28 +1,32 @@
...
     fn interects(&self, other: &Self) -> bool;
     fn contains(&self, other: &Self) -> bool;
 }
 
 pub struct AABB {
-    mins: Point3<f32>, //two points is all it takes to represent a box
-    maxs: Point3<f32>,
+    //two points is all it takes to represent a box
+    pub mins: Point3<f32>,
+    pub maxs: Point3<f32>,
```
Comment (M-Blomqvist) [inline pphysx3d/src/shapes/bounding_volume.rs:13]:
```suggestion
    mins: Point3<f32>,
    maxs: Point3<f32>,
```
Probably shouldn't allow changing the AABB's fields directly

File: pphysx3d/src/shapes/bounding_volume.rs [hunk 2]
```diff
@@ -1,28 +1,32 @@
...
 }
 
 impl AABB {
+    /// Creates a new axis-aligned bounding box with the given min points and max points
     pub fn new(mins: Point3<f32>, maxs: Point3<f32>) -> AABB {
         AABB { mins, maxs }
     }
```
Comment (M-Blomqvist) [inline pphysx3d/src/shapes/bounding_volume.rs:21]:
```suggestion
      pub fn mins(&self) -> Point3<f32>{
            self.mins
      }
      pub fn maxs(&self) -> Point3<f32>{
            self.maxs
      }
```
^see suggestion above

File: pphysx3d/src/shapes/bounding_volume.rs [hunk 3]
```diff
@@ -33,8 +37,10 @@ impl BoundingVolume for AABB {
     fn contains(&self, other: &Self) -> bool {
         na::partial_le(&self.mins, &other.mins) && na::partial_ge(&self.maxs, &other.maxs)
```
Comment (M-Blomqvist) [inline pphysx3d/src/shapes/bounding_volume.rs:38]:
```suggestion
        na::partial_le(&self.mins, &other.mins()) && na::partial_ge(&self.maxs, &other.maxs())
```

File: pphysx3d/src/shapes/bounding_volume.rs [hunk 4]
```diff
@@ -33,8 +37,10 @@ impl BoundingVolume for AABB {
...
     }
+    /// Checks if there is an intersection. Also returns true if self contains other
     fn interects(&self, other: &Self) -> bool {
-        na::partial_le(&self.mins, &other.maxs) && na::partial_ge(&self.maxs, &other.mins)
+        //a::partial_le(&self.mins, &other.maxs) && na::partial_ge(&self.maxs, &other.mins)
+        self.mins <= other.maxs && self.maxs >= other.mins
```
Comment (M-Blomqvist) [inline pphysx3d/src/shapes/bounding_volume.rs:43]:
```suggestion
        self.mins <= other.maxs() && self.maxs >= other.mins()
```

**Review by M-Blomqvist — CHANGES_REQUESTED** [review]
Don't know if Bounding Sphere also needs getters for its private fields

File: pphysx3d/src/shapes/bounding_volume.rs (hunk 3 above)
Comment (coffecup25) [reply pphysx3d/src/shapes/bounding_volume.rs:43]: shouldnt it be be &self.mins() aswell?

**Review by axellervik — APPROVED** [review]

File: pphysx3d/src/shapes/bounding_volume.rs (hunk 3 above)
Comment (isakvlarsson) [reply pphysx3d/src/shapes/bounding_volume.rs:43]: i don't think it matters if self uses the getter

**Comment by coffecup25** [comment]
> 
> 
> Don't know if Bounding Sphere also needs getters for its private fields

We'll notice if we need it later. For now I think it's fine

**Comment by coffecup25** [comment]
> 
> 
> Don't know if Bounding Sphere also needs getters for its private fields

We'll notice if it's needed later. It's fine for now

**Review by M-Blomqvist — APPROVED** [review]

**Pull request merged**